src/sql/workbench/browser/modelComponents/text.component.ts

```diff
@@ -26,14 +26,15 @@ import { IThemeService } from 'vs/platform/theme/common/themeService';
 	<div *ngIf="showDiv;else noDiv" style="display:flex;flex-flow:row;align-items:center;" [style.width]="getWidth()" [style.height]="getHeight()">
 	<p [title]="title" [ngStyle]="this.CSSStyles" [attr.role]="ariaRole" [attr.aria-hidden]="ariaHidden"></p>
 		<span #textContainer></span>
-		<p *ngIf="requiredIndicator" style="color:red;margin-left:5px;">*</p>
+		<span *ngIf="requiredIndicator" style="color:red;">*</span>
```

Comment:
Why are you changing this?

Reply:
Should we enclose this and p in a div? Otherwise, the star won't get the text styling

Reply:
I was modeling the structure for the required indicator in the [formContainer](https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/browser/modelComponents/formContainer.component.ts#L51).

Reply:
There was already the required indicator here though - why is this changing the element type and removing the margin? If you're making modifications to existing code please make sure that it's clear why a change is being done since this could have impacts across the application. Was there a bug you were seeing with how the indicator was displayed?

Reply:
The required indicator already existed here, however, the original html was structured so that custom css on the text would not apply when the required indicator is added.
Original:
```
<div>
  <p style=CUSTOM></p>
  <span>text</span>
  <p>*</p>
</div>
```

With these changes, we now have:
```
<p style=CUSTOM>
   <span>text</span>
  <span>*</span>
</p>
```

The text labels on this page have custom bolding and text sizes. When the required indicator is added, the original html structure does not allow the css to be applied, causing the Database Backup page on the sql migration to look like this:

![image](https://user-images.githubusercontent.com/14362577/125887283-11266e2d-52a9-463b-b143-34c34a1f66d2.png)

Reply:
I've searched other places across the application where the required indicator is being used with the text component, and these changes don't affect the look of the pages. 

Removing the margin here doesn't shift the element because with these changes, the text and * are both `<span>` elements under a parent `<p>` element. Previously, the `<span> text` and `<p> *`  were under a parent <div style="display:flex">, requiring the `<p> *`  element to have a margin.

I didn't find any scenarios in the code, however, in the case where the text has a description and required indicator, it will use the original structure/styles before this PR.

 Here are some examples:
![image](https://user-images.githubusercontent.com/14362577/125887379-6f37708d-2ad5-4a8d-88b6-e71439faac15.png)

![image](https://user-images.githubusercontent.com/14362577/125887396-fa261ba4-39b3-488e-9cfa-5d6e4ffc2f6e.png)